Source/JavaScriptCore/inspector/remote/cocoa/RemoteInspectorCocoa.mm

```diff
@@ -521,10 +524,10 @@ static bool canAccessWebInspectorMachPort()
     }
 
     if (auto presentingApplicationPID = target.presentingApplicationPID())
-        [listing setObject:identifierForPID(presentingApplicationPID.value()) forKey:WIRHostApplicationIdentifierKey];
+        [listing setObject:identifierForPID(presentingApplicationPID.value()).get() forKey:WIRHostApplicationIdentifierKey];
 
-    if (auto* connectionToTarget = m_targetConnectionMap.get(target.targetIdentifier()))
-        [listing setObject:connectionToTarget->connectionIdentifier() forKey:WIRConnectionIdentifierKey];
+    if (RefPtr connectionToTarget = m_targetConnectionMap.get(target.targetIdentifier()))
+        [listing setObject:connectionToTarget->protectedConnectionIdentifier().get() forKey:WIRConnectionIdentifierKey];
```

Comment:
i thought that it was the caller's responsibility to ensure that things are retained?
```mm
[listing setObject:RetainPtr { connectionToTarget->connectionIdentifier() }.get() forKey:WIRConnectionIdentifierKey];
```
ditto for all other callers of `protectedConnectionIdentifier`

Reply:
Here the caller is retaining the connectionIdentifier at the call site by calling `protectedConnectionIdentifier()` to get a RetainPtr on the stack.